src/main.rs

```diff
@@ -439,7 +440,38 @@ fn main() -> anyhow::Result<()> {
         );
         handle.join().expect("thread is not panicking")?;
     }
-    drop(toc_arc);
+    drop(search_runtime);
+    drop(update_runtime);
+    drop(general_runtime);
     drop(settings);
+    drop(wait_unwrap(toc_arc));
     Ok(())
 }
+
+/// Wait for the current thread to become a single referent of an [`Arc`] and returns it.
+///
+/// This methods move a value out from an [`Arc`] when there is only one reference left
+/// and it is safe to do so.
+///
+/// **CAUTION**: this method may create live lock if parallel threads are not going to finish
+/// and decrement [`Arc`] counter. Caller should initiate shutdown of all the threads holding the
+/// [`Arc`] before calling this method.
+fn wait_unwrap<T>(mut input: Arc<T>) -> T {
+    let start_time = Instant::now();
+    let mut notified = false;
+    loop {
```

Comment:
How many times do you go through that loop in general when running your test?

Reply:
In my setup I almost never saw _"Waiting for Arc to be dropped"_ message, so it's less than 100 iterations usually.